workflows/mhcquant.nf

```diff
@@ -227,35 +237,31 @@ workflow MHCQUANT {
         ch_comet_out_idxml_proceeding = ch_comet_out_idxml
     }
 
-    // Write this information to an tsv file
-    OPENMS_TEXTEXPORTER_COMET(ch_comet_out_idxml_proceeding)
-    ch_versions = ch_versions.mix(OPENMS_COMETADAPTER.out.versions.ifEmpty(null))
     // Index decoy and target hits
     OPENMS_PEPTIDEINDEXER(ch_comet_out_idxml_proceeding.join(ch_decoy_db))
     ch_versions = ch_versions.mix(OPENMS_PEPTIDEINDEXER.out.versions.ifEmpty(null))
-
-    //
-    // SUBWORKFLOW: Pre-process step for the quantification of the data
-    //
-    if (!params.skip_quantification) {
-        MAP_ALIGNMENT(
-            OPENMS_PEPTIDEINDEXER.out.idxml,
-            ch_clean_mzml_file
-        )
-        ch_proceeding_idx = MAP_ALIGNMENT.out.ch_proceeding_idx
-        ch_versions = ch_versions.mix(MAP_ALIGNMENT.out.versions.ifEmpty(null))
-    } else {
-        ch_proceeding_idx = OPENMS_PEPTIDEINDEXER.out.idxml
+    // Save indexed runs for later use to keep meta-run information. Sort based on file id
+    ch_proceeding_idx = OPENMS_PEPTIDEINDEXER.out.idxml
+            .map { meta, idxml ->
+                [[id:meta.sample + "_" + meta.condition], meta, [id:meta.id, file:idxml]]
+            }
+            .groupTuple()
             .map {
-                meta, raw ->
-                [[id:meta.sample + "_" + meta.condition, sample:meta.sample, condition:meta.condition, ext:meta.ext], raw]
+                merge_id, meta, idxml ->
+                // Create a custom comparator to sort based on id
+                def comparator = { a, b -> a.id <=> b.id }
+                [merge_id, meta.sort(comparator), idxml.sort(comparator).file]
             }
-            .groupTuple(by: [0])
-    }
 
+    ch_runs_to_merge = ch_proceeding_idx
```

Comment:
can this merge not be included with the last step that ends on line 254?

Reply:
changed it such that group_meta and a list of meta maps are saved